File: pset1-c/mario/mario.c
#include <stdio.h>
#include <cs50.h>

int get_height(string prompt);

// Print out a pyramid with '#' for the height specified by the user
int main(void)
{
    // Store int from user in a variable called 'height'
    int height = get_height("Height: ");

    // Loop over the number of rows
    for (int row = 1; row < height + 1; row++)
    {
        // For each row, first print out the blank spaces on the left
        for (int blank = 0; blank < (height - row); blank++)
        {
            printf(" ");
        }

        // Second, the hashes
        for (int hash = 0; hash < row; hash++)
        {
            printf("#");
        }

        // Third, a "gap" between pyramids (two spaces)
        printf("  ");

        // Finally, the adjacent pyramid
        for (int hash = 0; hash < row; hash++)
        {
            printf("#");
        }

        // Prepare for the next row
        printf("\n");
    }
}

// Propmt user for a positive integer
int get_height(string prompt)
{
    int n;

    // Repeat until user enters an integer between 1 and 8, inclusive
    do
    {
        n = get_int("%s", prompt);
    }
    while (n < 1 || n > 8);
    return n;
}

File: pset6-python/mario/more/mario.py
from cs50 import get_int


def main():
    """
    Print out a pyramid with '#' for the height chosen by the user
    """

    # Store int from user in a variable
    height = get_valid_int("Height: ")

    # Loop over the number of rows
    for row in range(1, height + 1):

        # Calculate and store the number of blanks and hashes at 'row'
        blanks = " " * (height - row)
        pyramid = "#" * row

        # Format row with f-string
        print(f"{blanks}{pyramid}  {pyramid}")


# Propmt user for a valid integer
def get_valid_int(prompt):
    while True:
        n = get_int(prompt)

        # Repeat until user inputs an int between 1-8, inclusive
        if n > 0 and n < 9:
            break
    return n


main()

